cartographer_rviz/cartographer_rviz/drawable_submap.h

```diff
@@ -95,11 +101,14 @@ class DrawableSubmap : public QObject {
   const int submap_index_;
 
   ::cartographer::common::Mutex mutex_;
+  ::rviz::DisplayContext* const context_;
   Ogre::SceneManager* const scene_manager_;
   Ogre::SceneNode* const scene_node_;
+  Ogre::SceneNode* const submap_node_;
```

Comment:
See `UpdateTransform()`. `submap_node_` (ex `scene_node_`) has its pose set to the slice pose; I would have to perform additional transformations to obtain the submap origin in the coordinate system of the submap slice. By having a `scene_node_` which has the same pose as the root node, I can just set the axes pose to the one written in `position_` and `orientation_`. Plus, I can hide both the submap and the axes by calling `scene_node_.setVisible()`.